staging/src/k8s.io/kubectl/pkg/cmd/describe/describe.go

```diff
@@ -88,19 +111,57 @@ type DescribeOptions struct {
 	genericclioptions.IOStreams
 }
 
-func NewCmdDescribe(parent string, f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
-	o := &DescribeOptions{
-		FilenameOptions: &resource.FilenameOptions{},
-		DescriberSettings: &describe.DescriberSettings{
-			ShowEvents: true,
-			ChunkSize:  cmdutil.DefaultChunkSize,
-		},
+// AddFlags registers flags for a cli
+func (flags *DescribeFlags) AddFlags(cmd *cobra.Command) {
+	cmdutil.AddFilenameOptionFlags(cmd, flags.FilenameOptions, "containing the resource to describe")
+	cmdutil.AddLabelSelectorFlagVar(cmd, &flags.Selector)
+	cmd.Flags().BoolVarP(&flags.AllNamespaces, "all-namespaces", "A", flags.AllNamespaces, "If present, list the requested object(s) across all namespaces. Namespace in current context is ignored even if specified with --namespace.")
+	cmd.Flags().BoolVar(&flags.DescriberSettings.ShowEvents, "show-events", flags.DescriberSettings.ShowEvents, "If true, display events related to the described object.")
+	cmdutil.AddChunkSizeFlag(cmd, &flags.DescriberSettings.ChunkSize)
+}
 
-		CmdParent: parent,
+// ToOptions converts from CLI inputs to runtime input
+func (flags *DescribeFlags) ToOptions(parent string, args []string) (*DescribeOptions, error) {
```

Comment:
Where is builder initialization? `o.NewBuilder = f.NewBuilder`

Reply:
So I'm directly assigning it here https://github.com/kubernetes/kubernetes/pull/112714/files#diff-22297d1a66ecc1d0164cf7d38d7e218aa9163d5f507233e631c9e1f0c65d258cR150, should I add it as a variable